lib/client.js

```diff
@@ -132,6 +135,14 @@ Client.prototype.getJwtFromUrl = function () {
 };
 
 /**
+ * Attempts to replace the JWT in the url with the new JWT passed in the function.
+ * @param {string} newJwt
+ */
+Client.prototype.replaceJWTInUrl = function (newJwt) {
+  window.location.hash = '/?jwt=' + newJwt;
```

Comment:
The jwt in the URL has to be obtained from the /sso endpoint during
redirect to the ID site.  It is one time use and cannot be used again, so
this patch will not work.
On May 6, 2016 07:57, "Edward Jiang" <notifications@github.com> wrote:

> In lib/client.js
> <https://github.com/stormpath/stormpath.js/pull/93#discussion_r62332540>:
>
> > @@ -132,6 +135,14 @@ Client.prototype.getJwtFromUrl = function () {
> >  };
> >
> >  /**
> > + * Attempts to replace the JWT in the url with the new JWT passed in the function.
> > + * @param {string} newJwt
> > + */
> > +Client.prototype.replaceJWTInUrl = function (newJwt) {
> > +  window.location.hash = '/?jwt=' + newJwt;
>
> We don't use the query string for this right now, and I don't think we
> should be putting the JWT in the query string at all anyways, for the same
> reasons as social login, so I would consider this a minor issue. We would
> know if we were changing how stormpath calls into ID Site, anyways.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/stormpath/stormpath.js/pull/93/files/35c20f2578fb7b6bb0e2ccb801f69f28fb3dc7e2#r62332540>
>